synapse/storage/database.py

```diff
@@ -2512,10 +2512,10 @@ def simple_search_list_txn(
         cls,
         txn: LoggingTransaction,
         table: str,
-        term: Optional[str],
+        term: str,
         col: str,
         retcols: Iterable[str],
```

Comment:
Aside: given that we're about to %s-substitute this in I wonder if it makes sense to make these LiteralStrings (at some point, anyway) in the future, C.f. https://github.com/matrix-org/synapse/issues/13473